pkg/scheduler/nodeinfo/node_info.go

```diff
@@ -471,11 +482,23 @@ func (n *NodeInfo) Clone() *NodeInfo {
 // VolumeLimits returns volume limits associated with the node
 func (n *NodeInfo) VolumeLimits() map[v1.ResourceName]int64 {
 	volumeLimits := map[v1.ResourceName]int64{}
+
 	for k, v := range n.AllocatableResource().ScalarResources {
 		if v1helper.IsAttachableVolumeResourceName(k) {
 			volumeLimits[k] = v
 		}
 	}
+
+	if n.csiNode != nil {
+		for i := range n.csiNode.Spec.Drivers {
+			d := n.csiNode.Spec.Drivers[i]
+			if d.Allocatable != nil && d.Allocatable.Count != nil {
+				k := v1.ResourceName(volumeutil.GetCSIAttachLimitKey(d.Name))
+				volumeLimits[k] = int64(*d.Allocatable.Count)
+			}
+		}
```

Comment:
This is kinda tricky. This will mean that, we may not be able to distinguish between if driver is installed on the node or volume limit is coming from `Node` object.